graphql_compiler/compiler/ir_sanity_checks.py

```diff
@@ -30,6 +30,43 @@ def sanity_check_ir_blocks_from_frontend(ir_blocks):
     _sanity_check_mark_location_preceding_optional_traverse(ir_blocks)
     _sanity_check_every_location_is_marked(ir_blocks)
     _sanity_check_coerce_type_outside_of_fold(ir_blocks)
+    _sanity_check_all_marked_locations_are_registered(ir_blocks, query_metadata_table)
+    _sanity_check_registered_locations_parent_locations(query_metadata_table)
+
+
+def _sanity_check_registered_locations_parent_locations(query_metadata_table):
```

Comment:
Another sanity check that occurs to me is one ensuring that only one `LocationInfo` has a `None` parent, if that's not covered elsewhere.